### ISSUE TITLE
Bug: Pressing enter in Search bar of repo tab scrolls to top and may refresh repos without doing any functionality

### ISSUE THREAD

**Issue by Deadreyo** [open-sauced/app#988]
### Describe the bug

![image](https://user-images.githubusercontent.com/52013393/224137050-1008604c-88c4-4f3c-95e9-f70f494e5562.png)
Clicking enter isn't supposed to work, yet it scrolls the page to top and may occasionally reload the repos.

This behavior should be removed.

### Steps to reproduce

Open Search bar in Explore's Repository Tab, write anything and click enter.

### Affected services

insights.opensauced.pizza

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [ ] I agree to follow this project's Contribution Docs

**Comment by 0-vortex** [comment]
We could (should IMHO) `tabindex` stuff :D

**Comment by doaortu** [comment]
I guess I can fix this, can I work on this issue?

and, I have suggestions: 
- add arrow up and down functionality to select item in auto complete
- show skeleton loading or spinner when autocomplete fetch data from api.

How's that sounds @0-vortex ?

**Comment by 0-vortex** [comment]
> I guess I can fix this, can I work on this issue?
> 
> 
> 
> and, I have suggestions: 
> 
> - add arrow up and down functionality to select item in auto complete
> 
> - show skeleton loading or spinner when autocomplete fetch data from api.
> 
> 
> 
> How's that sounds @0-vortex ?

Add tabindex in the mix and i'm sold 😁

**Comment by doaortu** [comment]
Hey thanks for responding.
I know little bit about tabIndex, and I've use it in another project to implement similar features (the arrow up and down button navigation) , basically I used tabIndex="-1" to every item in list to be able to navigate through code.

is that similar to what you want? or are you want different kind of tabIndex implementation? can you please elaborate more on that? just to make sure I got what you mean.

**Comment by 0-vortex** [comment]
> Hey thanks for responding. I know little bit about tabIndex, and I've use it in another project to implement similar features (the arrow up and down navigation) , basically I used tabIndex="-1" to every item in list to be able to navigate through code.
> 
> is that similar to what you want? or are you want different kind of tabIndex implementation? can you please elaborate more on that? just to make sure I got what you mean.

A good resource for all things accessibility is a11yproject: https://www.a11yproject.com/posts/how-to-use-the-tabindex-attribute/

**Comment by doaortu** [comment]
Okay I'll study that, thanks for the reference

**Comment by doaortu** [comment]
> A good resource for all things accessibility is a11yproject: https://www.a11yproject.com/posts/how-to-use-the-tabindex-attribute/

After I've read the article, I kinda get it supposed to be helping us navigate using ```Tab``` key, but I'm still confused about how this can help solve this particular issue:

> Clicking enter isn't supposed to work, yet it scrolls the page to top and may occasionally reload the repos.

I've search through internet and none mentioned about preventing enter or reload behaviour with tabindex. Or I misunderstood your statement?

> We could (should IMHO) `tabindex` stuff :D

Sorry for a lot of questions, but I'm just want it to get it right.

**Comment by 0-vortex** [comment]
> > A good resource for all things accessibility is a11yproject: https://www.a11yproject.com/posts/how-to-use-the-tabindex-attribute/
> 
> 
> 
> After I've read the article, I kinda get it supposed to be helping us navigate using ```Tab``` key, but I'm still confused about how this can help solve this particular issue:
> 
> 
> 
> > Clicking enter isn't supposed to work, yet it scrolls the page to top and may occasionally reload the repos.
> 
> 
> 
> I've search through internet and none mentioned about preventing enter or reload behaviour with tabindex. Or I misunderstood your statement?
> 
> 
> 
> > We could (should IMHO) `tabindex` stuff :D
> 
> 
> 
> Sorry for a lot of questions, but I just want it to avoid misunderstanding.

Enter also has to be fixed, however, before that we need to ensure we are focusing the right components. Pressing enter to submit is a desktop only feature, while navigating between different areas of the site is universal. 

More effort in a11y means the site will be available for people that are less able than the general user (think about screen readers, voice navigation, etc): that is more valuable than satisfying ease of use concern of normal users (having the enter work).

TL;DR: a11y > ux, when we are full 2.1 compliant we can make desktop-only improvements. Do the enter fix last, IMHO

**Comment by doaortu** [comment]
Thank you for the detailed clarification.

So, if I understand it correctly, we aim to be 2.1 compliant first.
we fix any accessibility problems, for now.

Do I need to fix accessibility problems on the whole site or on this page only?

Let me see what I can do with it. I'll update you if I'm able to do it and make a draft PR.

**Comment by 0-vortex** [comment]
> Thank you for the detailed clarification.
> 
> So, if I understand it correctly, we aim to be 2.1 compliant first. we fix any accessibility problems, for now.
> 
> Do I need to fix accessibility problems on the whole site or on this page only? I guess it would be huge PR to review if we fix the whole page?
> 
> Let me see what I can do with it. I'll update you if I'm able to do it and make a draft PR.

Just this part for this issue. Not sure just fixing tabindex will correct the enter, will have to check

**Comment by doaortu** [comment]
Okay I got it.
I'll first do some research how to make search-autocomplete component that respect the a11y / 2.1 compliant.
Earlier I've done some research on ```radix-ui```, turns out they don't have search-autocomplete component yet, will do research other library, maybe I'll find some insight how to implement it correctly.

Thank you for the guidance. I'll make a PR when I'm ready.

**Comment by takanome-dev** [comment]
@doaortu just checking in to see if you have questions. Also, feel free to ask for help if needed

**Comment by doaortu** [comment]
Hi, @takanome-dev , sorry for the late response.
Lately, I don't have much time to work on this issue.
I'm afraid I can't deliver it in a timely fashion.
Please feel free to unassign me if I'm interfering with the Milestone deadline.

**Comment by takanome-dev** [comment]
Thanks for the response, @doaortu. No rush or deadline for this issue, so take your time. If you're unable to work on it, feel free to unassign yourself. If anyone else is interested, please let me know. :slightly_smiling_face:

**Comment by sourabpramanik** [comment]
.take
